OpenEdXMobile/src/main/java/org/edx/mobile/view/MainTabsDashboardFragment.kt

```diff
@@ -0,0 +1,255 @@
+            AccountFragment::class.java,
+            resources.getString(R.string.profile_title),
+            R.drawable.ic_person, arguments
+        ) {
+            environment.analyticsRegistry.trackScreenViewEvent(
+                Analytics.Events.PROFILE_PAGE_VIEWED,
+                Analytics.Screens.PROFILE
+            )
+        })
+
+        return items
+    }
+
+    /**
+     * To handle the tab-selection of the ViewPager based on screen name [Screen] which may be
+     * sent through a deep link.
+     *
+     * @param bundle arguments
+     */
+    private fun handleTabSelection(bundle: Bundle?) {
+        bundle?.apply {
+            @ScreenDef val screenName = getString(Router.EXTRA_SCREEN_NAME)
+            if (screenName != null && !getBoolean(Router.EXTRA_SCREEN_SELECTED, false)) {
+                val fragmentItems = fragmentItems
+
+                fragmentItems.forEachIndexed { index, item ->
+                    if (shouldSelectFragment(item, screenName)) {
+                        binding.viewPager.currentItem = index
+                        return@forEachIndexed
+                    }
+                }
+                // Setting `EXTRA_SCREEN_SELECTED` to true, so that upon recreation of the fragment
+                // the tab defined in the deep link is not auto-selected again.
+                putBoolean(Router.EXTRA_SCREEN_SELECTED, true)
+            }
+        }
+    }
+
+    /**
+     * Determines if a tab fragment needs to be selected based on screen name.
+     *
+     * @param item       [FragmentItemModel] assigned to a tab.
+     * @param screenName Screen name param coming from [org.edx.mobile.deeplink.DeepLinkManager]
+     * @return           True if the specified tab needs to be selected, False otherwise
+     */
+    private fun shouldSelectFragment(
+        item: FragmentItemModel,
+        @ScreenDef screenName: String
+    ) = when {
+        screenName == Screen.PROGRAM && item.iconResId == R.drawable.ic_menu_book -> true
+        screenName == Screen.PROFILE && item.iconResId == R.drawable.ic_person -> true
+        screenName == Screen.DISCOVERY && item.iconResId == R.drawable.ic_search -> true
+        screenName == Screen.DISCOVERY_COURSE_DETAIL && item.iconResId == R.drawable.ic_search -> true
+        screenName == Screen.DISCOVERY_PROGRAM_DETAIL && item.iconResId == R.drawable.ic_search -> true
+        else -> false
+    }
+
+    @Subscribe
+    fun onEventMainThread(event: MoveToDiscoveryTabEvent) {
+        if (!environment.config.discoveryConfig.isDiscoveryEnabled) {
+            return
+        }
+        binding.viewPager.currentItem = 0
+        if (event.screenName != null) {
+            EventBus.getDefault().post(getNewInstance(event.screenName))
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        EventBus.getDefault().unregister(this)
+    }
+
+    fun onBackPressed(): Boolean {
+        if (tabBackStack.size > 1) {
```

Comment:
The back stack is not working correctly instead stack is adding the same instances again n again